src/lib/linear_model/logistic_regression.ts

```diff
@@ -0,0 +1,84 @@
+import {
+  mul,
+  randomUniform,
+  Rank,
+  round,
+  sigmoid,
+  Tensor,
+  Tensor1D,
+  tensor1d,
+  tensor2d
+} from '@tensorflow/tfjs';
+import {
+  checkNumFeatures,
+  inferShape,
+  tensorToArray,
+  to2DMatrix
+} from '../ops';
+import { Type1DMatrix, Type2DMatrix } from '../types';
+
+export class LogisticRegression {
+  private weights: Tensor1D;
+  private learningRate: number;
+
+  constructor(learningRate = 0.1) {
+    this.learningRate = learningRate;
+  }
+
+  public fit(
+    X: Type2DMatrix<number> | Type1DMatrix<number>,
+    y: Type1DMatrix<number>,
+    numIterations = 4000
+  ): void {
+    const xWrapped: Type2DMatrix<number> = to2DMatrix(X);
+    this.initWeights(xWrapped);
+    const tensorX = tensor2d(xWrapped);
+    const tensorY = tensor1d(y);
+
+    for (let i = 0; i < numIterations; ++i) {
+      const predictions: Tensor<Rank> = sigmoid(tensorX.dot(this.weights));
+
+      const gradient: Tensor<Rank> = mul(
```

Comment:
I prefer using `tf.mul` instead of importing `mul` explicitly to avoid any name shadowing